packages/wallet/src/redux/indirect-funding/player-a/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,238 @@
+    channelFundingStatus: fundingStates.SAFE_TO_DEPOSIT,
+    requestedTotalFunds: total,
+    requestedYourContribution: testScenarios.twoThree[0],
+    channelId: ledgerId,
+    ourIndex: PlayerIndex.A,
+  });
+  describe(whenActionArrives(actions.funding.FUNDING_RECEIVED_EVENT), () => {
+    const action = actions.funding.fundingReceivedEvent(defaults.ledgerId, total, total);
+    const updatedState = playerAReducer(walletState, action);
+
+    itTransitionToStateType(updatedState, states.WAIT_FOR_POST_FUND_SETUP_1);
+    itTransitionsChannelToStateType(
+      updatedState,
+      ledgerId,
+
+      channelStates.A_WAIT_FOR_POST_FUND_SETUP,
+    );
+    itTransitionsFundingToType(updatedState, ledgerId, fundingStates.CHANNEL_FUNDED);
+  });
+});
+
+describe(startingIn(states.WAIT_FOR_POST_FUND_SETUP_1), () => {
+  const { channelId, ledgerId } = defaults;
+  const walletState = { ...defaultWalletState };
+
+  walletState.indirectFunding = states.waitForPostFundSetup1({ channelId, ledgerId });
+  // Add the ledger channel to state
+  const ledgerChannelState = channelStates.aWaitForPostFundSetup({
+    ...ledgerChannelDefaults,
+    turnNum: 2,
+    channelId: ledgerId,
+  });
+  walletState.channelState.initializedChannels[ledgerId] = ledgerChannelState;
+  describe(whenActionArrives(actions.COMMITMENT_RECEIVED), () => {
+    const validateMock = jest.fn().mockReturnValue(true);
+    Object.defineProperty(SigningUtil, 'validCommitmentSignature', { value: validateMock });
+
+    const action = actions.commitmentReceived(
+      ledgerId,
+      WalletProcedure.IndirectFunding,
+      testScenarios.ledgerCommitments.postFundCommitment1,
+      '0x0',
+    );
+    const updatedState = playerAReducer(walletState, action);
+    itTransitionToStateType(updatedState, states.WAIT_FOR_LEDGER_UPDATE_1);
+    itTransitionsChannelToStateType(
+      updatedState,
+      ledgerId,
+
+      channelStates.WAIT_FOR_UPDATE,
+    );
+  });
+});
+
+describe(startingIn(states.WAIT_FOR_LEDGER_UPDATE_1), () => {
+  const { channelId, ledgerId } = defaults;
+  const walletState = { ...defaultWalletState };
+
+  walletState.indirectFunding = states.waitForLedgerUpdate1({ channelId, ledgerId });
+  // Add the ledger channel to state
+  const ledgerChannelState = channelStates.waitForUpdate({
+    ...ledgerChannelDefaults,
+    turnNum: testScenarios.ledgerCommitments.postFundCommitment1.turnNum,
+    channelId: ledgerId,
+  });
+  walletState.channelState.initializedChannels[ledgerId] = ledgerChannelState;
+  describe(whenActionArrives(actions.COMMITMENT_RECEIVED), () => {
+    const validateMock = jest.fn().mockReturnValue(true);
+    Object.defineProperty(SigningUtil, 'validCommitmentSignature', { value: validateMock });
+
+    const action = actions.commitmentReceived(
+      ledgerId,
+      WalletProcedure.IndirectFunding,
+      testScenarios.ledgerCommitments.ledgerUpdate1,
+      '0x0',
+    );
+    playerAReducer(walletState, action);
```

Comment:
Is the purpose of this describe block is to show that there's no run-time error?

Reply:
Yup, right now it only serves to make sure things don't blow up. Once we decide on what a "finished" state looks like this can be modified to do the proper check.